### ISSUE TITLE
Dynamic keyword requires `System.Runtime.CompilerServices.DynamicAttribute' to be defined

### ISSUE THREAD

**Issue by AIIX** [procore3d/giles#16]
Using Unity 5.5.0.b7 

Error: Assets/GILES/Code/Classes/Serialization/Converters/pb_ObjectContainerConverter.cs(23,43): error CS1980: Dynamic keyword requires `System.Runtime.CompilerServices.DynamicAttribute' to be defined. Are you missing System.Core.dll assembly reference?

**Comment by karl-** [comment]
Are you compiling with the complete .NET profile and not subset?

**Comment by AIIX** [comment]
Yes using .Net 2.0  and not the subset. I even reInstalled the final release of Unity 5.5 the error is still the same.  Also tried changing the project settings in Visual Studio to .Net 4.0 the error remains.  

Additionally I'm using unity3d 64bit on Windows10. Also tried the above in Unity 5.4 beta release on the Linux editor the error is the same.

**Comment by tomaszzackiewicz** [comment]
The same problem. I tried to add:
using System.Runtime.CompilerServices.DynamicAttribute;

But I get the error:
Assets/GILES/Code/Classes/Serialization/Converters/pb_ObjectContainerConverter.cs(5,39): error CS0234: The type or namespace name `DynamicAttribute' does not exist in the namespace `System.Runtime.CompilerServices'. Are you missing an assembly reference?

**Comment by karl-** [comment]
What platform are you targeting?

**Comment by tomaszzackiewicz** [comment]
Now, PC, 64bit on Windows10, but later I would like to use it on mobile too.

**Comment by karl-** [comment]
Looking at the code again I think you could safely just comment out then entirety of `pb_ObjectContainerConverter`.

**Comment by tomaszzackiewicz** [comment]
I have commented out all the code in the file. No errors and I can run the project, but I'm not sure if it won't prevent me from suing serialisation.

**Comment by karl-** [comment]
If I recall correctly that path was never actually used - I abandoned it for some reason, or at the least didn't find time to finish it.  So you should be able to serialize and deserialize without issue.

**Comment by tomaszzackiewicz** [comment]
Ok, thank you.

**Comment by samtulach** [comment]
Hello!
.NET 2.0 native not support **DynamicAttribute**, so [download this beta Unity build](https://forum.unity3d.com/threads/upgraded-mono-net-in-editor-on-5-5-0b9.438359/) with .NET 4.6. For me is working!

**Comment by SavedByZero** [comment]
I'm confused, what's the exact solution here?  I never had an issue using the dynamic keyword with .net 2.0 or .net 2.0 subset until recent 5.X releases, so why is a basic variable type suddenly breaking Unity?

**Comment by kevinw** [comment]
I'm running into this issue as well, with Unity 5.6.0f3, under both .NET 2.0 and .NET 2.0 Subset.

**Comment by SavedByZero** [comment]
The only thing that worked for me was replacing dynamic with object, and that's only because I'm a weird case with an xml reader that reads variable types as attributes and puts them into a dictionary.  In most cases, I would think dynamic and object are fairly different.
